test/e2e/major_upgrade_v2.go

```diff
@@ -16,13 +16,14 @@ import (
 )
 
 func MajorUpgradeToV2(logger *log.Logger) error {
-	numNodes := 4
-	upgradeHeight := int64(10)
-	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
for my own understanding, why don't we need the cancel any longer?

Reply:
The removal was suggested in the previous review process. Since `ctx.Background()` is tied to the main goroutine, it will remain valid until the entire test suite completes. We could put  `ctx.WithCancel()` back to tie the lifecycle explicitly to this particular test function, allowing for more precise resource management.

Also, the line `defer testNet.Cleanup(ctx)` ensures that all resources related to this specific test, including the Knuu instances, are cleaned up properly. So, using either approach is ok, but using `ctx.WithCancel()` provides more control over the test's lifecycle.